RHEcosystemAppEng/dbaas-operator
# Observability

## Description

This is initial PR which will add metrics for the RHODA operator and enabled the configuration of scraping the metrics for local prometheus, then push for remote write to observatorium as show in the below flow diagram.
<img width="1162" alt="Screen Shot 2022-07-20 at 2 08 06 PM" src="https://user-images.githubusercontent.com/41960152/180052484-d8c3124b-9dae-4ce2-a4e8-7da30289bbb9.png">

##Task Done:

1. Install the Observability operator by using reconciler code as done for other operators.
2. Once the Observability operator installed creating the observability operator [CR](https://github.com/RHEcosystemAppEng/dbaas-operator/blob/observability/controllers/reconcilers/providers_installation/observability-operator-cr.yaml) with this it created observability related components with in dbaas namespace. 
3. Enabled the Configuration for scraping the metrics from RHODA Operator, with this it will create ServiceMonirtor prometheus CR with label 'app: dbaas-prometheus'.
4. Added metrics for inventory status and request time duration from the provider operator response 
5. Added the connection metrics status and request time duration to the from the provider operator response 
6. Added the instance metrics status and request time duration to the from the provider operator response 
7. Added metrics for platform installation and total duration taken to complete the installation.
8. Review the ADR for more details here [1](https://docs.google.com/document/d/1ol1l8Ad77Ab2i2YOICgsle2xBff6U16Eu562CBb-W-k/edit) and [2 ](https://docs.google.com/document/d/1Tv9AV4smstEI139O_k2tXPUFG6fZb01xs9Nju2jIg4w/edit#)

## Verification Steps

1. Checkout the PR locally, and build the image and deploy the RHODA operator.
2. Verify the RHODA operator install the Observability operator along with other provider operators.
3. Verify the CR for Observability operator created successfully and added extra pods and services with in dbaas operator namesapce.
4. created the different objects for inventory, connection, instance, 
5. Query the prometheus to see the metrics related with dbaas operator exists locally.
6. Currently remote write configuration is hardcoded add the the details into observability operator CR and verify the metrics are pushing into remote observatorium.


## Type of change


## Checklist (Definition of Done)
